src/client/datascience/commands/commandRegistry.ts

```diff
@@ -66,26 +72,18 @@ export class CommandRegistry implements IDisposable {
         @inject(IJupyterServerUriStorage) private readonly serverUriStorage: IJupyterServerUriStorage,
         @inject(IJupyterVariables) @named(Identifiers.DEBUGGER_VARIABLES) private variableProvider: IJupyterVariables,
         @inject(UseVSCodeNotebookEditorApi) private readonly useNativeNotebook: boolean,
-        @inject(NotebookCreator) private readonly nativeNotebookCreator: NotebookCreator
+        @inject(NotebookCreator) private readonly nativeNotebookCreator: NotebookCreator,
+        @inject(IWorkspaceService) private readonly workspace: IWorkspaceService
     ) {
         this.disposables.push(this.serverSelectedCommand);
         this.disposables.push(this.notebookCommands);
         this.dataViewerChecker = new DataViewerChecker(configService, appShell);
+        if (!this.workspace.isTrusted) {
```

Comment:
This seems minor, but to be safe. It looks like might leave a gap for a possible double registration. If the workspace is trusted in between the constructor and register being called it looks like it would try to register twice. Should the event be hooked up at the point of register() being called?

Reply:
>  If the workspace is trusted in between the constructor and register being called it looks like it would try to reg

Cant happen as this is all part of the synchronous code flow (extension loading is synchronous).

Reply:
Sounds good as is then. Just want to avoid ever showing an unlocalized message to users.